### ISSUE TITLE
solver.load not working with StableBaseline

### ISSUE THREAD

**Issue by NansDarraillan** [airbus/scikit-decide#314]
## 🐛 Bug

Maybe this is not a bug but given StableBaseline models can be saved with solver.save, I would expect them to be loadable with solver.load. However, solver.load does not initialises well the StableBaseline solver correctly and leads to en exception when using the solver for training or rollout. (self.env is None)

## To Reproduce

1. Get to the example baselines_solvers.py
2.line 54: instead of 
`GymDomain.solve_with(solver, domain_factory, load_path="TEMP_Baselines")`
do:
```
solver.load("TEMP_Baselines", domain_factory)
GymDomain.solve_with(solver, domain_factory)
```
1.AssertionError
.venv\lib\site-packages\stable_baselines3\common\base_class.py", line 422, in _setup_learn       
    assert self.env is not None

## Expected behavior

I would expect the same result as with using load_path

## Environment

python 3.8.5, skdecide 0.9.8, Windows

**Comment by NansDarraillan** [comment]
I just realised: how can I do a rollout of a saved model without calling solve_with with the current behaviour?

**Comment by neo-alex** [comment]
Hello @NansDarraillan - sorry for the late reply!

I understand that your need is to reload a saved solver solution (e.g. to test it, without necessarily further training).

In that case, the code to be used is indeed [this one from the baselines_solvers.py example](https://github.com/airbus/scikit-decide/blob/26c7d550808c40010798197a8867a60722ba55b8/examples/baselines_solver.py#L53) (or similarly [this one for RLlib](https://github.com/airbus/scikit-decide/blob/26c7d550808c40010798197a8867a60722ba55b8/examples/rllib_solver_gym_domain.py#L43)). Despite the misleading naming (which we could try to improve in the future), calling `GymDomain.solve_with(solver, domain_factory, load_path="TEMP_Baselines")` with a given `load_path` will just reload the checkpoint **without** further training (unless you call `solver.solve(domain_factory)` again after), as indicated by the python docstring of the `load_path` argument: "load_path: The path to restore the solver state from (**if None, the solving process will be launched instead**)".

This may seem a bit counter-intuitive but the reason is that the library takes care of all necessary "auto-cast" inside the `Domain.solve_with` method (e.g. to enable a multi-agent solver to be automatically applicable to a single-agent domain), as opposed to the raw `solver.load` method which already expects the domain in the appropriate format (e.g. multi-agent domain for a multi-agent solver). This probably explains why your direct use of `solver.load` was not working in your case.

I hope this clarifies the issue you were facing and most importantly provides an easy solution you can use (even if we should reconsider some non-intuitive API naming).

**Comment by neo-alex** [comment]
Note: there was a bug specific to the Stable Baselines 3 wrapper which prevented to continue training after a reload; but it should be fixed now with the PR #358
